### PR TITLE
Add PHP 8.1 support, drop PHP 7.3

### PR DESCRIPTION
**Pull request by Slamdunk**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,10 +22,10 @@
         }
     ],
     "require": {
-        "php": "^7.3 || ^8.0"
+        "php": "~7.4.0 || ~8.0.0 || ~8.1.0"
```
Comment (Slamdunk) [inline composer.json:25]: I know it's too late to be restrictive on the upper bound, but better late then never, no?

Comment (Ocramius) [reply composer.json:25]: Aye: sad truth about `php-src`, heh.

@Potherca just FYI, PHP breaks compatibility in every damn minor release.

**Review by Ocramius — APPROVED** [review]
:ship:

**Comment by Ocramius** [comment]
Adjusted branch protection rules: :ship:

**Comment by Ocramius** [comment]
Thanks @Slamdunk!

File: composer.json (hunk 1 above)
Comment (Potherca) [reply composer.json:25]: Yeah. I've been through a few of those... been developing in PHP since 2002... 😢

**Pull request merged**